samples/core/enumopencl/main.cpp

```diff
@@ -17,132 +17,18 @@
-    {
-        if (param_value != NULL)
-        {
-            delete[] param_value;
-            param_value = NULL;
-        }
-    }
-
-    if (errorCode == CL_SUCCESS)
-    {
-        errorCode = clGetPlatformInfo(platformId, param_name, 0, NULL, &size);
-    }
-
-    if (errorCode == CL_SUCCESS)
-    {
-        if (size != 0)
-        {
-            param_value = new char[size];
-            if (param_value == NULL)
-            {
-                errorCode = CL_OUT_OF_HOST_MEMORY;
-            }
-        }
-    }
-
-    if (errorCode == CL_SUCCESS)
-    {
-        errorCode =
-            clGetPlatformInfo(platformId, param_name, size, param_value, NULL);
-    }
-
-    if (errorCode != CL_SUCCESS)
-    {
-        delete[] param_value;
-        param_value = NULL;
-    }
-
-    return errorCode;
-}
-
-static cl_int PrintPlatformInfoSummary(cl_platform_id platformId)
-{
-    cl_int errorCode = CL_SUCCESS;
-
-    char* platformName = NULL;
-    char* platformVendor = NULL;
-    char* platformVersion = NULL;
-
-    errorCode |= AllocateAndGetPlatformInfoString(platformId, CL_PLATFORM_NAME,
-                                                  platformName);
-    errorCode |= AllocateAndGetPlatformInfoString(
-        platformId, CL_PLATFORM_VENDOR, platformVendor);
-    errorCode |= AllocateAndGetPlatformInfoString(
-        platformId, CL_PLATFORM_VERSION, platformVersion);
-
-    printf("\tName:           %s\n", platformName);
-    printf("\tVendor:         %s\n", platformVendor);
-    printf("\tDriver Version: %s\n", platformVersion);
-
-    delete[] platformName;
-    delete[] platformVendor;
-    delete[] platformVersion;
-
-    platformName = NULL;
-    platformVendor = NULL;
-    platformVersion = NULL;
-
-    return errorCode;
-}
-
-static cl_int AllocateAndGetDeviceInfoString(cl_device_id device,
-                                             cl_device_info param_name,
-                                             char*& param_value)
-{
-    cl_int errorCode = CL_SUCCESS;
-    size_t size = 0;
-
-    if (errorCode == CL_SUCCESS)
-    {
-        if (param_value != NULL)
-        {
-            delete[] param_value;
-            param_value = NULL;
-        }
-    }
-
-    if (errorCode == CL_SUCCESS)
-    {
-        errorCode = clGetDeviceInfo(device, param_name, 0, NULL, &size);
-    }
-
-    if (errorCode == CL_SUCCESS)
-    {
-        if (size != 0)
-        {
-            param_value = new char[size];
-            if (param_value == NULL)
-            {
-                errorCode = CL_OUT_OF_HOST_MEMORY;
-            }
-        }
-    }
-
-    if (errorCode == CL_SUCCESS)
-    {
-        errorCode =
-            clGetDeviceInfo(device, param_name, size, param_value, NULL);
-    }
-
-    if (errorCode != CL_SUCCESS)
-    {
-        delete[] param_value;
-        param_value = NULL;
-    }
-
-    return errorCode;
+    printf("\tName:           %s\n",
+           platform.getInfo<CL_PLATFORM_NAME>().c_str());
```

Comment:
```suggestion
    std::cout << "\tName:           " << platform.getInfo<CL_PLATFORM_NAME>();
```
and put the `\n` before the next `\t`.